pkg/installmanager/installmanager.go

```diff
@@ -1689,3 +1702,48 @@ func getHomeDir() string {
 	}
 	return defaultHomeDir
 }
+
+func overrideCredentialsModeOnInstallConfig(manifestsDir, credentialsMode string) error {
+	clusterConfigPath := filepath.Join(manifestsDir, clusterConfigYAML)
+	clusterConfigBytes, err := ioutil.ReadFile(clusterConfigPath)
+	if err != nil {
+		return errors.Wrap(err, "error reading install config")
+	}
+
+	decoder := yml.NewYAMLOrJSONDecoder(bytes.NewReader(clusterConfigBytes), 4096)
+	clusterConfig := &corev1.ConfigMap{}
+	if err := decoder.Decode(clusterConfig); err != nil {
+		return errors.Wrap(err, "failed to decode cluster config")
+	}
+
+	data, ok := clusterConfig.Data[installConfigKeyName]
+	if !ok {
+		return fmt.Errorf("did not find key %s in data of cluster-config configmap", installConfigKeyName)
+	}
+
+	decoder = yml.NewYAMLOrJSONDecoder(bytes.NewReader([]byte(data)), 4096)
+	installConfig := &installertypes.InstallConfig{}
+	if err := decoder.Decode(installConfig); err != nil {
+		return errors.Wrap(err, "failed to decode install config")
+	}
+
+	installConfig.CredentialsMode = installertypes.CredentialsMode(credentialsMode)
```

Comment:
If we move the log line from above into this func, we could save the original value here and print both in the message. 
```go
m.log.WithField("origMode", origMode).WithField("newMode", credentialsMode).Info("override effective in-cluster credentials mode")
```
kind of thing. (Print it at the bottom of the func when we know we've succeeded.)

Reply:
I see the log line moved, but the old value isn't being logged. For that matter, it wouldn't be a terrible idea to have a check here like

```go
if installConfig.CredentialsMode == installertypes.CredentialsMode(credentialsMode) {
    m.log.WithField("credsMode", credentialsMode).Info("Not overriding already-set credentials mode")
    return nil
}
```